SalesforceCommerceCloud/pwa-kit
# @W-16778724 Optimize PLP Skeleton

# Description



Optimizes PLP page UI refresh when the shopper context is updated by ensuring only the pricing/promotions skeleton is displayed during the refetch rather than a skeleton for the entire `ProductTile`.

Initial Lighthouse Perf test results: https://salesforce.quip.com/GR0fAqgyDXMP#temp:C:VaK5c7366ee2f8640c48e626e370

# Types of Changes




> Breaking changes include:
>
> - Removing a public function or component or prop
> - Adding a required argument to a function
> - Changing the data type of a function parameter or return value
> - Adding a new peer dependency to `package.json`

# Changes

- PLP: instead of showing the entire `ProductTileSkeleton` during a refetch, only display a skeleton over the pricing and promotions

# How to Test-Drive This PR

- Checkout this branch and start the local server
- Navigate to http://localhost:3000/global/en-GB/category/womens-clothing-tops?sourceCode=instagram
- Verify only the pricing and promotions for each product displays a skeleton before the personalized data is rendered

# Checklists




## General


## Accessibility Compliance

You must check off all items in **one** of the follow two lists:


_or..._


## Localization
